homeassistant/components/evohome/coordinator.py

```diff
@@ -4,109 +4,143 @@
 
 from collections.abc import Awaitable
 from datetime import timedelta
+from http import HTTPStatus
 import logging
-from typing import TYPE_CHECKING, Any
+from typing import Any
 
-import evohomeasync as ev1
-from evohomeasync.schema import SZ_ID, SZ_TEMP
-import evohomeasync2 as evo
-from evohomeasync2.schema.const import (
+import evohomeasync as ec1
+import evohomeasync2 as ec2
+from evohomeasync2.const import (
     SZ_GATEWAY_ID,
     SZ_GATEWAY_INFO,
+    SZ_GATEWAYS,
     SZ_LOCATION_ID,
     SZ_LOCATION_INFO,
+    SZ_TEMPERATURE_CONTROL_SYSTEMS,
     SZ_TIME_ZONE,
+    SZ_USE_DAYLIGHT_SAVE_SWITCHING,
 )
+from evohomeasync2.schemas.typedefs import EvoLocStatusResponseT
 
-from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import CONF_LOCATION_IDX, DOMAIN, GWS, TCS, UTC_OFFSET
-from .helpers import handle_evo_exception
 
-if TYPE_CHECKING:
-    from . import EvoSession
+class EvoDataUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator for evohome integration/client."""
 
-_LOGGER = logging.getLogger(__name__.rpartition(".")[0])
-
-
-class EvoBroker:
-    """Broker for evohome client broker."""
-
-    loc_idx: int
-    loc: evo.Location
-    loc_utc_offset: timedelta
-    tcs: evo.ControlSystem
-
-    def __init__(self, sess: EvoSession) -> None:
-        """Initialize the evohome broker and its data structure."""
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        client_v2: ec2.EvohomeClient,
+        *,
+        name: str,
+        update_interval: timedelta,
+        location_idx: int,
+        client_v1: ec1.EvohomeClient | None = None,
+    ) -> None:
+        """Class to manage fetching data."""
+
+        super().__init__(
+            hass,
+            logger,
+            config_entry=None,
+            name=name,
+            update_interval=update_interval,
+        )
+
+        self.client = client_v2
+        self.client_v1 = client_v1
+
+        self.loc_idx = location_idx
+
+        # These will not be None after _async_setup())
+        self.loc: ec2.Location = None  # type: ignore[assignment]
+        self.tcs: ec2.ControlSystem = None  # type: ignore[assignment]
+
+        self.data: EvoLocStatusResponseT = None  # type: ignore[assignment]
+        self.temps: dict[str, float | None] = {}
 
-        self._sess = sess
-        self.hass = sess.hass
+        self._first_refresh_done = False  # get schedules only after first refresh
 
-        assert sess.client_v2 is not None  # mypy
+    # our version of async_config_entry_first_refresh()...
+    async def async_first_refresh(self) -> None:
```

Comment:
But why again? The normal one would only raise `ConfigEntryNotReady` and we could catch that and return `False` in the init function

Reply:
But `async_config_entry_first_refresh()` responds in a way that indicated I shouldn't use it unless **evohome** had config flow, which it does not:
```
  'homeassistant.helpers.frame',
  30,
  "Detected that integration 'evohome' uses "
  '`async_config_entry_first_refresh`, which is only supported for '
  'coordinators with a config entry at '
  'homeassistant/components/evohome/coordinator.py, line 76: await '
  'super().async_config_entry_first_refresh(). This will stop working in '
  'Home Assistant 2025.11, please create a bug report at '
  'https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+evohome%22',
```

Reply:
The relevant line of code: https://github.com/home-assistant/core/blob/c4411914c2e906105b765c00af5740bd0880e946/homeassistant/helpers/update_coordinator.py#L290-L295

Reply:
> The normal one would _only_ raise ConfigEntryNotReady and...

TL;DR: the 'only' (my emphasis) isn't 100% true.